### ISSUE TITLE
Inconsistent results reproducing pretrain_pbmc.ipynb

### ISSUE THREAD

**Issue by wwwm072991-rgb** [GET-Foundation/get_model#62]
Hi authors,
﻿
Thanks for the great work and the detailed tutorials!
﻿
I am currently trying to reproduce the pretraining process using the provided pretrain_pbmc.ipynb and the exact same example dataset. However, I noticed a very significant discrepancy in the validation metrics (specifically masked_pearson) between the tutorial's output and my local run, even though the configuration and data are identical.

Expected Behavior (from the tutorial):

<img width="316" height="516" alt="Image" src="https://github.com/user-attachments/assets/32fa240b-953d-4a01-b453-7ef736ecaf2c" />

Actual Behavior (my local run):

<img width="5365" height="1459" alt="Image" src="https://github.com/user-attachments/assets/94941040-21f8-4271-bec3-1fa75832af98" />

My Configuration File:
cfg:
  run:
    project_name: pretrain_pbmc
    run_name: pretrain_pbmc_from_pretrain_lora
    use_wandb: false
  type: region
  stage: fit
  assembly: hg38
  eval_tss: true
  log_image: true
  model:
    _target_: get_model.model.model.GETRegionPretrain
    cfg:
      num_regions: 900
      num_motif: 283
      embed_dim: 768
      num_layers: 12
      num_heads: 12
      dropout: 0.1
      output_dim: 283
      flash_attn: false
      pool_method: mean
      region_embed:
        num_features: 283
        embed_dim: 768
      encoder:
        num_heads: 12
        embed_dim: 768
        num_layers: 12
        drop_path_rate: 0.1
        drop_rate: 0
        attn_drop_rate: 0
        use_mean_pooling: false
        flash_attn: false
      head_mask:
        in_features: 768
        out_features: 283
      mask_token:
        embed_dim: 768
        std: 0.02
      loss:
        components:
          masked:
            _target_: torch.nn.MSELoss
            reduction: mean
        weights:
          masked: 1.0
      metrics:
        components:
          masked:
          - pearson
          - mse
          - r2
  machine:
    codebase: /home/xf2217/Projects/get_model/
    data_path: /home/xf2217/Projects/get_data/
    output_dir: /home/model
    num_devices: 1
    num_workers: 2
    batch_size: 8
    fasta_path: ???
  dataset:
    zarr_path: /home/sample_data/pbmc10k_multiome.zarr
    celltypes: memory_b,cd14_mono,gdt,cd8_tem_1,naive_b,mait,intermediate_b,cd4_naive,cd8_tem_2,cd8_naive,cd4_tem,cd4_tcm,cd16_mono,nk,cdc,treg
    transform: null
    quantitative_atac: false
    sampling_step: 100
    num_region_per_sample: 200
    leave_out_chromosomes: null
    leave_out_celltypes: cd8_tem_1
    mask_ratio: 0.5
  training:
    save_ckpt_freq: 1
    epochs: 20
    warmup_epochs: 10
    accumulate_grad_batches: 1
    clip_grad: null
    use_fp16: false
    log_every_n_steps: 25
    val_check_interval: 1.0
    add_lr_monitor: false
    save_every_n_epochs: null
    checkpoint_monitor: val_masked_loss
  optimizer:
    lr: 0.0001
    min_lr: 0.0001
    weight_decay: 0.05
    opt: adamw
    opt_eps: 1.0e-08
    opt_betas:
    - 0.9
    - 0.999
  finetune:
    resume_ckpt: null
    pretrain_checkpoint: false
    checkpoint: /home/GET/checkpoint-799.pth
    strict: true
    model_key: model
    use_lora: true
    lora_checkpoint: null
    rename_config:
      encoder.head.: head_mask.
      encoder.region_embed: region_embed
      region_embed.proj.: region_embed.embed.
      encoder.cls_token: cls_token
    layers_with_lora:
    - region_embed
    - encoder
    patterns_to_freeze: []
    patterns_to_drop: []
    additional_checkpoints: []
  task:
    test_mode: interpret
    gene_list: MYC,SOX10,SOX2,RET
    layer_names:
    - region_embed
    mutations: null

Could you help clarify why there is such a massive gap in the masked_pearson metric (0.39 vs 0.77) under the same settings?Any insights or guidance would be greatly appreciated. Thank you!
